src/wallet.cpp

```diff
@@ -4062,6 +4062,9 @@ void CWallet::CreateAutoMintTransaction(const CAmount& nMintAmount, CCoinControl
 // CWallet::AutoZeromint() gets called with each new incoming block
 void CWallet::AutoZeromint()
 {
+    if (!pwalletMain)
```

Comment:
Shouldn't this be checked in main.cpp before this function is called?

```
@@ -3251,7 +3251,7 @@ void static UpdateTip(CBlockIndex* pindexNew)
     chainActive.SetTip(pindexNew);
 
     // If turned on AutoZeromint will automatically convert PIV to zPIV
-    if (pwalletMain->isZeromintEnabled ())
+    if (pwalletMain && pwalletMain->isZeromintEnabled ())
         pwalletMain->AutoZeromint ();
 
     // New best block
```
Only this change and disabling both staking and masternode fixed the problem for me.

Reply:
Quite right, the better place to check for `pwalletMain` is in `main.cpp`. updated and force pushed to reflect the change.